api/wallet.go

```diff
@@ -25,125 +74,343 @@ func scanAmount(amount string) (types.Currency, bool) {
-// current wallet.
-func (srv *Server) walletMergeHandler(w http.ResponseWriter, req *http.Request) {
-	// Scan the wallet file.
-	err := srv.wallet.MergeWallet(req.FormValue("walletfile"))
+// walletHistoryHandlerGETaddr handles a GET request to
+// /wallet/history/$(addr).
+func (srv *Server) walletHistoryHandlerGETaddr(w http.ResponseWriter, req *http.Request, addr types.UnlockHash) {
+	addrHistory, err := srv.wallet.AddressHistory(addr)
 	if err != nil {
-		writeError(w, err.Error(), http.StatusBadRequest)
+		writeError(w, "error after call to /wallet/history/$(addr): "+err.Error(), http.StatusBadRequest)
 		return
 	}
-	writeSuccess(w)
+	writeJSON(w, addrHistory)
+}
+
+// walletHistoryHandler handles all API calls to /wallet/history
+func (srv *Server) walletHistoryHandler(w http.ResponseWriter, req *http.Request) {
+	// Check for a vanilla call to /wallet/history.
+	if req.URL.Path == "/wallet/history" && req.Method == "GET" || req.Method == "" {
+		srv.walletHistoryHandlerGET(w, req)
+	}
+
+	// The only remaining possibility is a GET call to /wallet/history/$(addr);
+	// check that the method is correct.
+	if req.Method != "GET" && req.Method != "" {
+		writeError(w, "unrecognized method in call to /wallet/history", http.StatusBadRequest)
+		return
+	}
+
+	// Parse the address from the url and call the GETaddr Handler.
+	jsonAddr := "\"" + req.URL.Path[len("/wallet/history/"):] + "\""
+	var addr types.UnlockHash
+	err := addr.UnmarshalJSON([]byte(jsonAddr))
+	if err != nil {
+		writeError(w, "error after call to /wallet/history: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	srv.walletHistoryHandlerGETaddr(w, req, addr)
+}
+
+// walletSeedHandlerGET handles a GET request to /wallet/seed.
+func (srv *Server) walletSeedHandlerGET(w http.ResponseWriter, req *http.Request) {
+	dictionary := mnemonics.DictionaryID(req.FormValue("dictionary"))
+	if dictionary == "" {
+		dictionary = mnemonics.English
+	}
+
+	// Get the primary seed information.
+	primarySeed, progress, err := srv.wallet.PrimarySeed()
+	if err != nil {
+		writeError(w, "error after call to /wallet/seed: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	primarySeedStr, err := modules.SeedToString(primarySeed, dictionary)
+	if err != nil {
+		writeError(w, "error after call to /wallet/seed: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Get the list of seeds known to the wallet.
+	allSeeds := srv.wallet.AllSeeds()
+	var allSeedsStrs []string
+	for _, seed := range allSeeds {
+		str, err := modules.SeedToString(seed, dictionary)
+		if err != nil {
+			writeError(w, "error after call to /wallet/seed: "+err.Error(), http.StatusBadRequest)
+			return
+		}
+		allSeedsStrs = append(allSeedsStrs, str)
+	}
+	writeJSON(w, WalletSeedGET{
+		PrimarySeed:        primarySeedStr,
+		AddressesRemaining: int(modules.PublicKeysPerSeed - progress),
+		AllSeeds:           allSeedsStrs,
+	})
+}
+
+// walletSeedHandlerPUT handles a PUT request to /wallet/seed.
```

Comment:
you're supposed to reference the api documentation.

Maybe it's the wrong approach, but the idea was that we could make the api documentation more consistent with the code behavior if you did all the coding directly from the api docs.